crates/router/src/connector/checkout/transformers.rs

```diff
@@ -347,12 +350,14 @@ impl TryFrom<types::PaymentsResponseRouterData<PaymentsResponse>>
                 item.data.request.capture_method,
             )),
             response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(item.response.id),
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.id.clone()),
                 redirection_data,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response.reference.unwrap_or(item.response.id),
```

Comment:
Should this be an error ? Because it might break logic.

Reply:
We'll try to find our reference id, if it's not available we'll send connector's id. This shouldn't break the flow anyway because we are not using it anywhere except sending the value in payments response.